File: backend/app/schemas/audit.py
"""
Audit log schemas.

Spec reference: §30 Audit & Traceability
"""

from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class AuditLogResponse(BaseModel):
    id: str
    action: str
    entity_type: str
    entity_id: str
    entity_version: int | None
    performed_by: str | None
    summary: str
    request_snapshot: dict[str, Any] | None
    result_snapshot: dict[str, Any] | None
    metadata: dict[str, Any] | None
    correlation_id: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class AuditLogListItem(BaseModel):
    id: str
    action: str
    entity_type: str
    entity_id: str
    performed_by: str | None
    summary: str
    correlation_id: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/audit.ts
export interface AuditListItem {
  audit_id: string
  action: string
  entity_type: string
  entity_id: string
  entity_version?: number
  performed_by?: string
  summary: string
  correlation_id: string
  created_at: string
}

export interface AuditDetail extends AuditListItem {
  request_snapshot?: Record<string, unknown>
  result_snapshot?: Record<string, unknown>
  rule_versions?: Array<{ policy_id: string; version: number }>
  metadata?: Record<string, unknown>
}
